### PR TITLE
Added ability to configure usage of translator

### LINKED ISSUE #93: translator.default detection issue
The fix for #91 wasn't great, because sometimes, this service will not be available at loading time. Proposed solution:

1. Introduce a new `use_translator` configuration entry, defaulting to `false`
2. Prepend a configuration that enables it the translation is enabled in the configuration under `framework`
3. Make sure the configuration in 1. has priority over it.

### PR DESCRIPTION
**Pull request by curtchn**
Fixes #93

### CODE REVIEW

File: src/Bridge/Symfony/DependencyInjection/Configuration.php [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+class Configuration implements ConfigurationInterface
+{
+    public function getConfigTreeBuilder()
+    {
+        $treeBuilder = new TreeBuilder();
+        $rootNode = $treeBuilder->root('greg0ire_enum');
+
+        $rootNode
+            ->children()
+                ->booleanNode('use_translator')->defaultTrue()->end()
```
Comment (greg0ire) [inline src/Bridge/Symfony/DependencyInjection/Configuration.php]: Please add a call to `info()` here, with a description of this configuration node

File: src/Bridge/Symfony/DependencyInjection/Greg0ireEnumExtension.php [hunk 2]
```diff
@@ -6,28 +6,51 @@
...
+    public function prepend(ContainerBuilder $container)
+    {
+        $configs = $container->getExtensionConfig('framework');
+        $translatorEnabled = false;
+        foreach ($configs as $config) {
+            // the last one should win
+            if (isset($config['translator']['enabled'])) {
+                $translatorEnabled = (bool) $config['translator']['enabled'];
+            } elseif (isset($config['translator']['fallbacks'])) {
+                $translatorEnabled = (bool) $config['translator']['fallbacks'];
```
Comment (greg0ire) [inline src/Bridge/Symfony/DependencyInjection/Greg0ireEnumExtension.php]: Shouldn't it be just `$translatorEnabled = true` here or does symfony disable the translator if you do this:

```yaml
framework:
    translator:
        fallbacks: []
```

File: src/Bridge/Symfony/DependencyInjection/Greg0ireEnumExtension.php [hunk 3]
```diff
@@ -6,28 +6,51 @@
...
+
+    public function prepend(ContainerBuilder $container)
+    {
+        $configs = $container->getExtensionConfig('framework');
+        $translatorEnabled = false;
+        foreach ($configs as $config) {
+            // the last one should win
+            if (isset($config['translator']['enabled'])) {
+                $translatorEnabled = (bool) $config['translator']['enabled'];
+            } elseif (isset($config['translator']['fallbacks'])) {
```
Comment (greg0ire) [inline src/Bridge/Symfony/DependencyInjection/Greg0ireEnumExtension.php]: I think this is 👍 because `enabled` probably has priority over `fallbacks`

**Comment by Soullivaneuh** [comment]
IMHO, we should move the logic to a compiler pass to test.

You should not propose translation usage on your bundle, it depends on the framework-bundle.

**Comment by greg0ire** [comment]
> IMHO, we should move the logic to a compiler pass to test.

@curtchn has been working on doing exactly that lately, it should come soon :)

**Pull request closed without merging**